src/main/java/com/example/management/models/Movie.java

```diff
@@ -0,0 +1,95 @@
+package com.example.management.models;
+
+import javax.persistence.*;
+import java.util.*;
+/**
+ * Movie
+ */
+@Entity
+@Table(name = "movie")
+public class Movie extends BaseEntity{
+
+    private String name;
+    private String description;
+    private int year;
+    private Date releaseDay;
+    private String image;
+
+    @ManyToMany
+    @JoinTable(
+            name = "movie_type",
+            joinColumns = @JoinColumn(name = "movie_id"),
+            inverseJoinColumns = @JoinColumn(name = "type_id")
+    )
+    private Set<Types> typesSet = new HashSet<>();
+
+    @OneToMany(
+            mappedBy = "movie",
+            cascade = CascadeType.ALL,
+            orphanRemoval = true
+    )
+    private List<Rate> rates = new ArrayList();
+
+    private Movie(Builder builder) {
+        //id = builder.id;
+        this.name = builder.name;
+        this.description = builder.description;
+        this.year = builder.year;
+        this.releaseDay = builder.releaseDay;
+        this.image = builder.image;
+        this.typesSet = builder.typesSet;
+//        this.rates = builder.rates;
+    }
+
+    public static class Builder {
+//         Required parameters
+//        private int id;
+        private String name;
+        private String description;
+        private int year;
+        private Date releaseDay;
+        private String image;
+        private Set<Types> typesSet = new HashSet<>();
+        private ArrayList rates= new ArrayList();
+
+        public Builder(String name) {

```

Comment:
I think that mandatory attributes expected to be in the constructor of Builder. And because every movie has a name so I have put it on the constructor above. Is there any misunderstood made by me?

Reply:
It's not mandatory to change, this still correct. For my idea is to have default constructor without depending on specific attribute 
And I think it's a title bit better e.g:

Movie.builder().name('').description(""). However, the current state is ok.